File: code/problems/max/maximum.c
#include <stdio.h>

int maximum(int first,int second,int third)
{
	if (first>=second)
	{
		if (first>=third)
		{
			return first;
		}
		return third;
	}
	else
	{
		if (second>=third)
		{
			return second;
		}
		return third;
	}
}
int max_of_array(int[],int);
int main()
{
	printf("%d %d %d %d\n",maximum(1,2,3),maximum(4,3,2),maximum(4,5,3),maximum(1,2,2));
	int arr[]={1,2,3,5,4};
	printf("%d\n",max_of_array(arr,5));
	return 0;
}
int max_of_array(int arr[],int n)
{
	int maximum=arr[0];
	for(int i=1;i<n;i++)
	{
		if (arr[i]>maximum)
		{
			maximum=arr[i];
		}
	}
	return maximum;
}

File: code/problems/max/Maximum.java
import java.util.Collections;
import java.util.Arrays;
public class Maximum{
	public static void main(String[] args){
		System.out.println(maximum(1,2,3)+" "+maximum(3,2,1)+" "+maximum(1,3,2)+" "+maximum(2,1,2));
		Integer[] arr={1,5,4,3,2};
		//System.out.println(maxOfArray(arr));
		System.out.println(Math.max(1,2));//it takes only two arguments
		System.out.println(Collections.max(Arrays.asList(arr)));
	}
	public static int maximum(int first,int second,int third){
		if (first>=second && first>=third){
			return first;
		}
		if (second>=first && second>=third){
			return second;
		}
		return third;
	}
	public static int maxOfArray(int[] arr){
		int maximum=arr[0];
		for (int element:arr){
			if (element>maximum){
				maximum=element;
			}
		}
		return maximum;
	}
}
